config/routes.rb

```diff
@@ -1,8 +1,13 @@
 Rails.application.routes.draw do
-  get 'users/show'
-
+  post 'follow' => 'users#follow'
+  delete 'unfollow' => 'users#unfollow'
   get 'profile/:username' => 'users#show', as: :profile
-  resources :posts
+  resources :posts do
+    member do
+      post 'favorite'
+      delete 'unfavorite'
+    end
+  end
   devise_for :users, :controllers => {registrations: 'users/registrations'}
   root 'posts#index'
   # The priority is based upon order of creation: first created -> highest priority.
```

Comment:
Again, feel free to remove these auto-generated comments. Most of the time, if we need help, we'll look at the docs, instead of these comments.